tests/capture/test_capture_qnode.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for capturing a qnode into jaxpr.
+"""
+from functools import partial
+
+# pylint: disable=protected-access
+import pytest
+
+import pennylane as qml
+from pennylane.capture.capture_qnode import _get_qnode_prim
+
+qnode_prim = _get_qnode_prim()
+
```

Comment:
Throughout this test file, it might be nice to check that directly executing a QNode still works with `qml.capture.enabled()=True` (not only via the manual capturing + `eval_jaxpr` route).

Reply:
adding some more "normal execution" tests inside each of the test cases.